src/BloomExe/Edit/PageListView.cs

```diff
@@ -142,18 +142,32 @@ public new bool Enabled
 			set { _thumbNailList.Enabled = value; }
 		}
 
+		private bool _processingClickHandler = false;
 		private void _addPageButton_Click(object sender, EventArgs e)
 		{
-			if (_model.CanAddPages)
+			if (_processingClickHandler)
+				return;
+			try
 			{
-				_model.ShowAddPageDialog();
+				_processingClickHandler = true;
+				if (_model.CanAddPages)
+				{
+					_model.ShowAddPageDialog();
+					// BL-2743 Make sure the dialog is actually showing before we allow another click
+					// (which will then be caught by the js and ignored)
+					Application.DoEvents();
```

Comment:
Application.DoEvents() is a last-resort solution, as it leads to hard to find bugs.

In this case, would a simpler solution would be to just "de-bounce" this button by remembering when the last click was, and ignoring the click if it is within, say, a second?